usaspending_api/awards/v2/filters/sub_award.py

```diff
@@ -224,14 +224,13 @@ def recip_string_parse(recipient_string):
             queryset = queryset.filter(q) if q else queryset
 
         # add "naics_codes" (column naics) after NAICS are mapped to subawards
-        elif key in ("program_numbers", "contract_pricing_type_codes"):
-            filter_to_col = {
-                "program_numbers": "cfda_number",
-                "contract_pricing_type_codes": "type_of_contract_pricing",
-            }
-            in_query = [v for v in value]
-            if len(in_query) != 0:
-                queryset &= SubawardView.objects.filter(**{"{}__in".format(filter_to_col[key]): in_query})
+        elif key in ("contract_pricing_type_codes"):
+            if len(value) != 0:
+                queryset &= SubawardView.objects.filter(type_of_contract_pricing__in=value)
+
+        elif key == "program_numbers":
+            if len(value) != 0:
+                queryset &= SubawardView.objects.filter(award__transactions__cfda_number__in=value)
```

Comment:
This doesn't appear to be working as expected. When I test locally I am seeing these results:
```
{
    "limit": 60,
    "results": [
        {
            "internal_id": "4100078275",
            "prime_award_internal_id": 44333000,
            "Sub-Award ID": "4100078275",
            "prime_award_generated_internal_id": "ASST_NON_1705PA5ADM_7530"
        },
        {
            "internal_id": "4100078275",
            "prime_award_internal_id": 44333000,
            "Sub-Award ID": "4100078275",
            "prime_award_generated_internal_id": "ASST_NON_1705PA5ADM_7530"
        },
        {
            "internal_id": "4100078275",
            "prime_award_internal_id": 44333000,
            "Sub-Award ID": "4100078275",
            "prime_award_generated_internal_id": "ASST_NON_1705PA5ADM_7530"
        },
        ...
   ],
}
```
The ORM code here is causing LEFT OUTER JOIN to `awards` and `transaction_search` which results in duplicate entries being returned for each Subaward